habitat-lab/habitat/tasks/nav/nav.py

```diff
@@ -1289,16 +1403,543 @@ def step(
+        self._max_wait_duration = self._config.max_wait_duration
+
+    def step(self, *args: Any, **kwargs: Any):
+        r"""Update ``_metric``, this method is called from ``Env`` on each
+        ``step``.
+        """
+        xyt_waypoint = np.array([self._forward_step_size, 0.0, 0.0])
+        return self._step_rel_waypoint(
+            xyt_waypoint, self._config.max_wait_duration, *args, **kwargs
+        )
+
+    @property
+    def action_space(self):
+        return EmptySpace()
+
+
+@registry.register_task_action
+class TurnLeftWaypointAction(WaypointAction):
+    name: str = "turn_left_waypoint"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._turn_angle = self._config.turn_angle
+        self._max_wait_duration = self._config.max_wait_duration
+
+    def step(self, *args: Any, **kwargs: Any):
+        r"""Update ``_metric``, this method is called from ``Env`` on each
+        ``step``.
+        """
+        xyt_waypoint = np.array([0.0, 0.0, self._turn_angle])
+        return self._step_rel_waypoint(
+            xyt_waypoint, self._config.max_wait_duration, *args, **kwargs
+        )
+
+    @property
+    def action_space(self):
+        return EmptySpace()
+
+
+@registry.register_task_action
+class TurnRightWaypointAction(WaypointAction):
+    name: str = "turn_right_waypoint"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._turn_angle = self._config.turn_angle
+        self._max_wait_duration = self._config.max_wait_duration
+
+    def step(self, *args: Any, **kwargs: Any):
+        r"""Update ``_metric``, this method is called from ``Env`` on each
+        ``step``.
+        """
+        xyt_waypoint = np.array([0.0, 0.0, -self._config.turn_angle])
+        return self._step_rel_waypoint(
+            xyt_waypoint, self._config.max_wait_duration, *args, **kwargs
+        )
+
+    @property
+    def action_space(self):
+        return EmptySpace()
+
+
+@registry.register_task_action
+class LookUpDiscreteToVelocityAction(WaypointAction):
+    name: str = "look_up_discrete_to_velocity"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._turn_angle = self._config.turn_angle
+        self._max_wait_duration = self._config.max_wait_duration
+
+    def step(self, *args: Any, **kwargs: Any):
+        r"""Update ``_metric``, this method is called from ``Env`` on each
+        ``step``.
+        """
+        xyt_waypoint = np.array([0.0, 0.0, 0.0])
+        delta_camera_pitch_angle = self._turn_angle
+        return self._step_rel_waypoint(
+            xyt_waypoint,
+            self._config.max_wait_duration,
+            delta_camera_pitch_angle,
+            *args,
+            **kwargs,
+        )
+
+    @property
+    def action_space(self):
+        return EmptySpace()
+
+
+@registry.register_task_action
+class LookDownDiscreteToVelocityAction(WaypointAction):
```

Comment:
The name of this class is rather confusing, what does it do? Don't we already have a discrete action for looking up/down ? What is the difference between the two implementations ?

Reply:
Thank you! This is for making sure that we can do sim2real transfer. This function takes the discrete action command and produces the velocity command for the motor so that we can mimic the real world hardware. The original one does not have such capability and just teleport the camera.

Reply:
Would it be beneficial to just change the existing action to produce a velocity command?
Or maybe rename the old one to LookDownInstantaneousAction ?